js/packages/openinference-core/test/trace/contextAttributes.test.ts

```diff
@@ -130,3 +169,58 @@ describe("context.with multiple attributes", () => {
     );
   });
 });
+
+describe("getContextAttributes", () => {
+  let contextManager: ContextManager;
+  beforeEach(() => {
+    contextManager = new AsyncHooksContextManager().enable();
+    context.setGlobalContextManager(contextManager);
+  });
+  afterEach(() => {
+    context.disable();
+  });
+
+  it("should get all attributes off of the context", () => {
+    const variables = {
+      name: "world",
+    };
+    context.with(
+      setMetadata(
+        setSession(
+          setPromptTemplate(context.active(), {
+            template: "hello {name}",
+            variables,
+            version: "V1.0",
+          }),
+          { sessionId: "session-id" },
+        ),
+        { key: "value" },
+      ),
+      () => {
+        const attributes = getAttributesFromContext(context.active());
+        expect(attributes).toStrictEqual({
+          [PROMPT_TEMPLATE_TEMPLATE]: "hello {name}",
+          [PROMPT_TEMPLATE_VARIABLES]: JSON.stringify(variables),
+          [PROMPT_TEMPLATE_VERSION]: "V1.0",
+          [SESSION_ID]: "session-id",
+          [METADATA]: JSON.stringify({ key: "value" }),
+        });
+      },
+    );
```

Comment:
Can you add to the tests that these attributes get un-set?

Reply:
get un-set from the context or un-set from where? Right now getAttributes from context doesn't do any un-setting although we could, that wouldn't allow you to read the context attributes more than once though which seems wrong